### ISSUE TITLE
Styling bugs in admin-side menus on mobile

### ISSUE THREAD

**Issue by aviannamiller** [omeka/omeka-s#2368]
There are some styling bugs on the user and search menus designed for mobile viewports:
- <img width="252" height="448" alt="User menu. There are two small buttons at the top with poor contrast, styled with dark red text on dark blue backgrounds. Below, a logout button, which fills half of the screen in height." src="https://github.com/user-attachments/assets/d1c2dcc1-877a-43f6-aee4-aa1dfeaa3215" />

  - On the user menu, `.user-show.o-icon-user` and `.user-settings` buttons only have a 2:1 contrast ratio (#a91919 text/icon on #222933 backgrounds). Contrast here should be at least 4.5:1.
  - Something has gone awry with the way size is calculating for the logout button (namely height), or at least the paragraph element it is contained within.
- <img width="252" height="447" alt="Search bar menu. The search submit button is floated up to the right, away from the search bar and out of alignment with an advanced options button." src="https://github.com/user-attachments/assets/c307b283-9dec-4c06-b6ad-1969fb2df6d4" /> <img width="252" height="448" alt="Search bar menu, with advanced options expanded. The search submit button remains out of alignment." src="https://github.com/user-attachments/assets/2a4288ef-0404-41ff-95f4-e5099fae4dd2" />

  - On the search menu, the `#search-submit` button is way out of alignment with other buttons on the search bar. This is true whether the advanced options are open or closed, contrary to a similar issue with the same button discussed in https://github.com/omeka/omeka-s/issues/2333#issuecomment-3392502602.
